src/scripts/filtering.js

```diff
@@ -37,17 +59,52 @@ const filterPosts = async posts => {
 };
 
 export const main = async () => {
-  ({ ownPosts, duplicatePosts, filterText, filterTags, filterCws } = await getOptions('filtering'));
+  ({ ownPosts, hideShares, duplicatePosts, filterText, filterTags, filterCws } = await getOptions('filtering'));
   [filterText, filterTags, filterCws] = [filterText, filterTags, filterCws].map(list => list.trim() === '' ? false : list.split(',').map(str => str.trim().toLowerCase()));
   if (!ownPosts && duplicatePosts === 'disabled' && !filterText && !filterTags && !filterCws) return;
   if (window.location.pathname === `/${activeProject.handle}`) {
     ownPosts = false;
     duplicatePosts = false;
   }
+
+  doHideShares = hideShares && validHideSharesLocation(location.href)
+  if (doHideShares) {
+
+    const spacer = $(`<div class="flex-1" id=${spacerId}>&nbsp;</div>`)
+
+    const label = $(`
+    <label class="font-bold pl-4 text-sidebarText" id="${labelId}">
+      <span class="pr-2">hide shares</span>
+      <input class="h-6 w-6 rounded-lg border-2 border-foreground bg-notWhite text-foreground focus:ring-foreground pl-4" type="checkbox" id="${checkboxId}">
+    </label>
+  `)
+
+    const parent = $parent()
+    parent.append(spacer)
+    parent.append(label)
+    parent.attr('class', 'mb-2 flex flex-col items-center lg:flex-row')
+
+    $('#' + checkboxId).on("change", () => {
+      $('main').toggleClass(hideSharesClass)
+    })
+  }
+
+  mutationManager.start(parentSelector, () => {
```

Comment:
non-anonymous function for mutation callback

Reply:
does my new solution work?? we need to have a closure over the label variable in that function

Reply:
seems to work!